### PR TITLE
rebuild errors

### PR DESCRIPTION
**Pull request by batrudinych**

### CODE REVIEW

**Review by tarcisiozf — APPROVED** [review]

File: lib.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+'use strict'
+
+const _ = require('lodash')
+
+const getErr = err => _.isString(err) ? new Error(err) : err
+
+const buildErr = err => err ? new Error(err.message || err) : null
```
Comment (vigan-abd) [inline lib.js:7]: isn't this hiding the original error trace?

File: lib.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+'use strict'
```
Comment (vigan-abd) [inline lib.js:1]: this file is more like util, I would prefer to have lib/util.js instead

File: lib.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+const _ = require('lodash')
+
+const getErr = err => _.isString(err) ? new Error(err) : err
```
Comment (vigan-abd) [inline lib.js:5]: both utils are similar so I would opt in to have just getErr and adjust  `if (isPromise) {...` section to use getError when error is not nil

**Review by vigan-abd — CHANGES_REQUESTED** [review]

File: lib/utils.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+'use strict'
+
+const buildErr = err => err ? new Error(err.message || err) : null
```
Comment (vigan-abd) [inline lib/utils.js:3]: so based on here https://github.com/bitfinexcom/bfx-facs-grc/pull/17#discussion_r1409608557 my thought is this:
const buildErr = (err) => err ? (err instanceof Error ? err : new Error(err)) : null

Comment (vigan-abd) [reply lib/utils.js:3]: well we always lost error trace so this is techinicaly an improvement

**Review by vigan-abd — APPROVED** [review]

**Pull request merged**